app/models/page/DashboardPage.scala

```diff
@@ -107,7 +107,11 @@ object DashboardPage {
 
   private def activeApplication(user: CachedData)(implicit request: RequestHeader, lang: Lang): DashboardPage = {
     val firstStep = if (RoleUtils.activeUserWithApp(user)) ProgressActive else ProgressInactive
-    val secondStep = if (DisplayOnlineTestSectionRole.isAuthorized(user)) ProgressActive else ProgressInactive
+    val secondStep = if (DisplayOnlineTestSectionRole.isAuthorized(user) || FastPassReceivedRole.isAuthorized(user)) {
```

Comment:
@Lansalo 's refactor also fixes this on the dashboard page by introducing a viewmodel